### DISCUSSION TITLE
Code Organization

### DISCUSSION THREAD

**Discussion by eriicafes** [eriicafes/revision-study-client#2] [General]
Let's talk about file structure and conventions to follow

**Comment by princecodes247** [comment]
Ok

**Comment by eriicafes** [comment]
**Components**
All truly reusable pure components that accept props to control their look and behavior
- **ui**
Ui components usable across the whole application eg `Button.tsx`
- **[feature]**
Feature components for specific use cases eg. `question` for components around question entity eg. `QuestionItem.tsx`

**Layouts**
Page layout components eg. `HomeLayout.tsx`

**Pages**
Top level components that must be wrapped by a Layout component 

**Api**
React query hooks for data fetching

**Containers**
Smart components that maintain their own internal state and may interact with external state, containers may contain their own components that are not reusable elsewhere in a `partials` subfolder eg. `NavBar.tsx`

**Context**
For shared data across all pages and their components, not needed for state management eg `Config.tsx`

**Hooks**
Reusable react hooks eg `use-local-storage.tsx`

**Interfaces**
Entity Interface and types

**Helpers**
Exported utility functions

**Reply by princecodes247** [reply]
So UI is a folder in components?

**Reply by princecodes247** [reply]
Please Give examples so it's clear

**Reply by princecodes247** [reply]
Just like you did for the Layout

**Reply by eriicafes** [reply]
Yes exactly ui components are Button, Input, Select, etc

**Reply by eriicafes** [reply]
> Just like you did for the Layout
> 
> 

Layouts is it's own folder outside of components

**Comment by eriicafes** [comment]
Other style guides for react 

- Named exports for components except nextjs page components that must be default export
- Absolute imports as in tsconfig.json

**Reply by eriicafes** [reply]
Please feel free to add yours
